### DISCUSSION TITLE
using pest_derive in my WASM module

### DISCUSSION THREAD

**Discussion by adam-cyclones** [pest-parser/pest#616] [General]
I am working in a Node.js based WASM project, I would like tell Pest about my grammar, unfortunately the grammar cannot be opened for whatever reason. I can pass files into my lex() function as a WASM export.

I know that the Pest website runs a WASM version with dynamic grammar so this most be possible, although I cant find the Rust src's for it.

How do I derive grammar as a string but not using `grammar` or `grammar_inline`.

Note that I am a Rust beginner.

**Comment by max-sixty** [comment]
Following up on this — is anyone familiar with using `pest-derive` in a wasm target? We've used pest quite successfully in PRQL, but now we want to put it on the web (like Pest's, ironically). Do we need to rewrite the grammar in `pest_vm`, like Pest's site? https://github.com/pest-parser/site/blob/master/Cargo.toml

(NB: we don't need a dynamic grammar like Pest's site uses, just a standard grammar, but in wasm)

ref https://github.com/max-sixty/prql/issues/175

**Comment by CAD97** [comment]
pest_derive reads the file system to find the actual grammar file when you use `#[grammar = "path"]`. If you use `#[grammar_inline]`, then no file system access is done.

However, if you just want to use a pest generated parser in wasm, any normal pest parser should work fine, as this filesystem access is done at build time.

**Comment by max-sixty** [comment]
Thanks @CAD97 . I'll have another go on my end — sounds like `pest` should be compatible with this.

**Comment by adam-cyclones** [comment]
Wow I halted my project because of this but Id love to give it another shot, Im still watching this from the sidelines
